kola/tests/podman/podman.go

```diff
@@ -0,0 +1,352 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package podman
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/net/context"
+
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/lang/worker"
+	"github.com/coreos/mantle/platform"
+	"github.com/coreos/mantle/util"
+)
+
+// init runs when the package is imported and takes care of registering tests
+func init() {
+	register.Register(&register.Test{
+		Run:         podmanBaseTest,
+		ClusterSize: 1,
+		Name:        `podman.base`,
+		Distros:     []string{"rhcos"},
+	})
+	register.Register(&register.Test{
+		Run:              podmanWorkflow,
+		ClusterSize:      1,
+		Name:             `podman.workflow`,
+		ExcludePlatforms: []string{"qemu"}, // Requires internet for pulling nginx
+		Distros:          []string{"rhcos"},
+	})
+	register.Register(&register.Test{
+		Run:         podmanNetworkTest,
+		ClusterSize: 2,
+		Name:        `podman.network`,
+		Distros:     []string{"rhcos"},
+	})
+}
+
+// simplifiedContainerPsInfo represents a container entry in podman ps -a
+type simplifiedContainerPsInfo struct {
+	Id     string `json:"id"`
+	Image  string `json:"image"`
+	Status string `json:"status"`
+}
+
+// simplifiedPsInfo represents the results of podman ps -a
+type simplifiedPsInfo struct {
+	containers []simplifiedContainerPsInfo
+}
+
+// simplifiedPodmanInfo represents the results of podman info
+type simplifiedPodmanInfo struct {
+	Store struct {
+		GraphDriverName string `json:"GraphDriverName"`
+		GraphRoot       string `json:"GraphRoot"`
+	}
+}
+
+func getSimplifiedPsInfo(c cluster.TestCluster, m platform.Machine) (simplifiedPsInfo, error) {
+	target := simplifiedPsInfo{}
+	psJson, err := c.SSH(m, `sudo podman ps -a --format json`)
+
+	if err != nil {
+		return target, fmt.Errorf("could not get info: %v", err)
+	}
+
+	err = json.Unmarshal(psJson, &target.containers)
+
+	if err != nil {
+		return target, fmt.Errorf("could not unmarshal info %q into known json: %v", string(psJson), err)
+	}
+	return target, nil
+}
+
+// Returns the result of podman info as a simplifiedPodmanInfo
+func getPodmanInfo(c cluster.TestCluster, m platform.Machine) (simplifiedPodmanInfo, error) {
+	target := simplifiedPodmanInfo{}
+
+	pInfoJson, err := c.SSH(m, `sudo podman info --format json`)
+	if err != nil {
+		return target, fmt.Errorf("Could not get info: %v", err)
+	}
+
+	err = json.Unmarshal(pInfoJson, &target)
+	if err != nil {
+		return target, fmt.Errorf("Could not unmarshal info %q into known JSON: %v", string(pInfoJson), err)
+	}
+	return target, nil
+}
+
+// make a podman container out of binaries on the host
+func genPodmanContainer(c cluster.TestCluster, m platform.Machine, name string, binnames []string) {
```

Comment:
This seems to be the same as a sub-section of the [`genContainer`](https://github.com/coreos/mantle/blob/master/kola/tests/crio/crio.go#L216-L220) function in the cri-o tests. This should be split out into the `kola/tests/util` package and referenced in both locations to prevent code duplication.